notebook/tests/selenium/utils.py

```diff
@@ -251,10 +259,28 @@ def new_window(browser, selector=None):
 
 def shift(browser, k):
     """Send key combination Shift+(k)"""
-    ActionChains(browser)\
-        .key_down(Keys.SHIFT).send_keys(k).key_up(Keys.SHIFT).perform()
+    trigger_keystrokes(browser, "shift-%s"%k)
 
 def ctrl(browser, k):
     """Send key combination Ctrl+(k)"""
-    ActionChains(browser)\
-        .key_down(Keys.CONTROL).send_keys(k).key_up(Keys.CONTROL).perform()
+    trigger_keystrokes(browser, "control-%s"%k)
+
+def trigger_keystrokes(browser, *keys):
+    """ Send the keys in sequence to the browser.
+    Handles following key combinations
+    1. with modifiers eg. 'control-alt-a', 'shift-c'
+    2. just modifiers eg. 'alt', 'esc'
+    3. non-modifiers eg. 'abc'
+    Modifiers : http://seleniumhq.github.io/selenium/docs/api/py/webdriver/selenium.webdriver.common.keys.html
+    """
+    for each_key_combination in keys:
+        keys = each_key_combination.split('-')
+        if len(keys) > 1:  # key has modifiers eg. control, alt, shift
+            modifiers_keys = list(map(lambda x:getattr(Keys, x.upper()), keys[:-1]))
```

Comment:
`list(map(lambda` is a pretty good sign that a list comprehension would be clearer. ;-)

Reply:
Ah, muscle memory of using maps in Python2. 
You are right, "list comprehension" would be much clearer.